### PR TITLE
spelling corrections using en_US dictionary

### PR DESCRIPTION
**Pull request by tkphd**
Ran `ispell` from `emacs` against every `rst` and `py` file in FiPy. Corrected the errors it found, more or less.

---
This change is [Reviewable](https://reviewable.io/reviews/usnistgov/fipy/594)

### CODE REVIEW

File: documentation/BUILDBOT.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ http://build.cmi.kent.edu:8010/
 Modifying the configuration
 ---------------------------
 
-The buildbot configuration file is hosted in a Git repository. To clone the
+The Buildbot configuration file is hosted in a Git repository. To clone the
```
Comment (wd15) [inline documentation/BUILDBOT.rst]: We could delete this file. Shall I open an issue or do you want to do it as part of this pull request.

File: documentation/CODESPEED.rst [hunk 2]
```diff
@@ -16,7 +16,7 @@ Using Codespeed 0.8.1
 
 To download it, go to https://github.com/tobami/codespeed and use the
 downloads button on the right side to get the latest stable
```
Comment (wd15) [inline documentation/CODESPEED.rst]: Likewise, this file can go as well since we've never actually used codespeed.

File: INSTALLATION.rst [hunk 3]
```diff
@@ -328,7 +328,7 @@ environment manager.
 
 .. attention::
 
-   :term:`PySparse` and :term:`FiPy` are not presently included in Canopy,
```
Comment (guyer) [inline INSTALLATION.rst]: 1. [Pysparse](http://pysparse.sourceforge.net) is not consistent with [PySparse's](http://pysparse.sourceforge.net/introduction.html) own spelling.

1. CamelCaseRocks

QED. Existing spelling is correct

File: documentation/CREDITS.rst [hunk 4]
```diff
@@ -13,7 +13,7 @@ Contributors
     volume techniques, and level set treatments.
```
Comment (wd15) [inline documentation/CREDITS.rst]: Should we get rid of this as well and just point people at the Github profiles? This is so out of date. Another issue?

File: README.rst [hunk 5]
```diff
@@ -237,7 +237,7 @@ or a
 .. _StackOverflow: http://stackoverflow.com/questions/tagged/fipy
 
 .. |GitHub|        image:: https://img.shields.io/github/contributors/usnistgov/fipy.svg
-.. _Github:        https://github.com/usnistgov/fipy
```
Comment (guyer) [inline README.rst]: LeaveCamelCaseAlone!!!

File: documentation/BUILDBOT.rst [hunk 6]
```diff
@@ -12,7 +12,7 @@ http://build.cmi.kent.edu:8010/
 Modifying the configuration
 ---------------------------
 
-The buildbot configuration file is hosted in a Git repository. To clone the
```
Comment (guyer) [inline documentation/BUILDBOT.rst]: This pull request should wait to have #591 accepted and then merge `develop`

File: documentation/RESOURCES.rst [hunk 7]
```diff
@@ -1,20 +1,19 @@
 Computational Resources
```
Comment (guyer) [inline documentation/RESOURCES.rst]: This file removed in #591

File: fipy/terms/faceTerm.py [hunk 8]
```diff
@@ -19,7 +19,7 @@
  # United States Code this software is not subject to copyright
  # protection, and this software is considered to be in the public domain.
  # FiPy is an experimental system.
- # NIST assumes no responsibility whatsoever for its use by whatsever for its use by
+ # NIST assumes no responsibility whatsoever for its use by whatsoever for its use by
```
Comment (wd15) [inline fipy/terms/faceTerm.py]: That don't make no sense.

File: documentation/numerical/equation.rst [hunk 9]
```diff
@@ -21,7 +21,7 @@ Examples of such systems are wide ranging, but include problems that
 exhibit a combination of diffusing and reacting species, as well as
 such diverse problems as determination of the electric potential in
 heart tissue, of fluid flow, stress evolution, and even the
-Schroedinger equation.
```
Comment (guyer) [inline documentation/numerical/equation.rst]: If you're going to be this pedantic, why misspell the reference label?

File: documentation/sphinxext/docscrape.py [hunk 10]
```diff
@@ -463,7 +463,7 @@ def __str__(self):
 class ClassDoc(NumpyDocString):
     def __init__(self,cls,modulename='',func_doc=FunctionDoc,doc=None):
         if not inspect.isclass(cls):
-            raise ValueError("Initialise using a class. Got %r" % cls)
```
Comment (guyer) [inline documentation/sphinxext/docscrape.py]: This isn't actually our code, although I'm not positive of either the canonical source, nor if this is still the way we should be using it

File: documentation/BUILDBOT.rst (hunk 1 above)
Comment (tkphd) [reply documentation/BUILDBOT.rst]: Deleted.

File: examples/cahnHilliard/sphereDaemon.py [hunk 11]
```diff
@@ -10,7 +10,7 @@
 class SphereDaemon(MayaviDaemon):
 
     def view_data(self):
-        """Sets up the mayavi pipeline for the visualization.
```
Comment (guyer) [inline examples/cahnHilliard/sphereDaemon.py]: [mayavi](https://github.com/enthought/mayavi) used to be [MayaVi](http://mayavi.sourceforge.net/), but now it's [mayavi](https://github.com/enthought/mayavi), except when sometimes it's [Mayavi](https://github.com/enthought/mayavi/blob/master/README.rst#installation). *le sigh*

File: documentation/CREDITS.rst (hunk 4 above)
Comment (tkphd) [reply documentation/CREDITS.rst]: Beyond the scope of this PR. Ticket filed.

File: examples/phase/impingement/mesh40x1.py [hunk 12]
```diff
@@ -256,3 +256,8 @@
     exec(fipy.tests.doctestPlus._getScript())
 
     raw_input('finished')
+
+#  LocalWords:  WarrenPolycrystal ModularVariable et al Acta Materialia mPhiVar
```
Comment (guyer) [inline examples/phase/impingement/mesh40x1.py]: Hrmmmm...

Not sure I approve of cluttering up our code with this sort of thing. There's no way to add a dictionary to the repo?

File: fipy/terms/faceTerm.py (hunk 8 above)
Comment (tkphd) [reply fipy/terms/faceTerm.py]: Hmm, no, it does not.

File: ez_setup.py [hunk 13]
```diff
@@ -100,7 +100,7 @@ def do_download():
     try:
         import pkg_resources
     except ImportError:
-        return do_download()
```
Comment (guyer) [inline ez_setup.py]: This isn't our code

File: documentation/CODESPEED.rst (hunk 2 above)
Comment (tkphd) [reply documentation/CODESPEED.rst]: Dropped.

File: fipy/meshes/gmshMesh.py [hunk 14]
```diff
@@ -101,8 +101,8 @@ class MeshExportError(GmshException):
 def gmshVersion(communicator=parallelComm):
     """Determine the version of Gmsh.
 
-    We can't trust the generated msh file for the correct version number, so
-    we have to retrieve it from the gmsh binary.
+    We can't trust the generated .msh file for the correct version number, so
+    we have to retrieve it from the Gmsh binary.
```
Comment (guyer) [inline fipy/meshes/gmshMesh.py]: The binary is invoked as `gmsh`, not "Gmsh"

**Comment by tkphd** [comment]
This new GitHub Reviews interface is an improvement. I like it.

File: fipy/tests/doctestPlus.py [hunk 15]
```diff
@@ -243,7 +243,7 @@ def testmod(m=None, name=None, globs=None, verbose=None,
     default, no extra globals are used.  This is new in 2.4.
 
     Optional keyword arg "verbose" prints lots of stuff if true, prints
-    only failures if false; by default, it's true iff "-v" is in sys.argv.
+    only failures if false; by default, it's true iff. "-v" is in sys.argv.
```
Comment (guyer) [inline fipy/tests/doctestPlus.py]: [No](https://en.wikipedia.org/wiki/If_and_only_if)

File: INSTALLATION.rst (hunk 3 above)
Comment (tkphd) [reply INSTALLATION.rst]: 1. I disagree, [PySparse](http://pysparse.sourceforge.net/) is not consistent with [Pysparse](http://pysparse.sourceforge.net/introduction.html)'s own spelling.

  | Count | PySparse | Pysparse |
  | ---       | ---             | ---             |
  | [index](http://pysparse.sourceforge.net/index.html) | 0 | 13 |
  | [intro](http://pysparse.sourceforge.net/introduction.html) | 5 | 8 |
2. Take that up with the [Pysparse devs](https://sourceforge.net/projects/pysparse/support).

File: README.rst (hunk 5 above)
Comment (tkphd) [reply README.rst]: Unclear what you mean here. GitHub has more camels than Github, with the added benefit of existing.

File: documentation/numerical/equation.rst (hunk 9 above)
Comment (tkphd) [reply documentation/numerical/equation.rst]: True.

File: examples/phase/impingement/mesh40x1.py (hunk 12 above)
Comment (tkphd) [reply examples/phase/impingement/mesh40x1.py]: Oh, that's an erroneous include, sorry. `emacs` must have barfed on exit.

File: ez_setup.py (hunk 13 above)
Comment (tkphd) [reply ez_setup.py]: It's in our repository.

File: INSTALLATION.rst (hunk 3 above)
Comment (wd15) [reply INSTALLATION.rst]: It's Pysparse in headings and PySparse in sentences. Everyone knows that. It's important that we get this one correct.

Comment (tkphd) [reply INSTALLATION.rst]: I thought it was PySparse when sunny, Pysparse when overcast?

Comment (guyer) [reply INSTALLATION.rst]: There are no PySparse devs

File: README.rst (hunk 5 above)
Comment (guyer) [reply README.rst]: Apologies. Given the malice you committed with `PySparse` I somehow inverted the order of the diff.

File: INSTALLATION.rst (hunk 3 above)
Comment (tkphd) [reply INSTALLATION.rst]: [PEP 8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names) suggests the correct name is "pysparse." When used as a proper noun, English grammar prefers "Pysparse."

**Pull request merged**